dspace-server-webapp/src/test/java/org/dspace/app/rest/RegistrationRestRepositoryIT.java

```diff
@@ -0,0 +1,90 @@
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RegistrationRestRepositoryIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private RegistrationDataDAO registrationDataDAO;
+
+    @Test
+    public void findByTokenTestExistingUserTest() throws Exception {
+        String email = eperson.getEmail();
+        createTokenForEmail(email);
+        RegistrationData registrationData = registrationDataDAO.findByEmail(context, email);
+
+        getClient().perform(get("/api/eperson/registrations/search/findByToken")
+                                .param("token", registrationData.getToken()))
+                   .andExpect(status().isOk())
+                   .andExpect(
+                       jsonPath("$", Matchers.is(RegistrationMatcher.matchRegistration(email, eperson.getID()))));
+
+        email = "newUser@testnewuser.com";
+        createTokenForEmail(email);
+        registrationData = registrationDataDAO.findByEmail(context, email);
+
+        getClient().perform(get("/api/eperson/registrations/search/findByToken")
+                                .param("token", registrationData.getToken()))
+                   .andExpect(status().isOk())
+                   .andExpect(
+                       jsonPath("$", Matchers.is(RegistrationMatcher.matchRegistration(email, null))));
+
+        registrationDataDAO.delete(context, registrationData);
+
+    }
+
+    @Test
+    public void findByTokenTestNewUserTest() throws Exception {
+        String email = "newUser@testnewuser.com";
+        createTokenForEmail(email);
+        RegistrationData registrationData = registrationDataDAO.findByEmail(context, email);
+
+        getClient().perform(get("/api/eperson/registrations/search/findByToken")
+                                .param("token", registrationData.getToken()))
+                   .andExpect(status().isOk())
+                   .andExpect(
+                       jsonPath("$", Matchers.is(RegistrationMatcher.matchRegistration(email, null))));
+
+        registrationDataDAO.delete(context, registrationData);
+    }
+
+    @Test
+    public void findByTokenNotExistingTokenTest() throws Exception {
+        getClient().perform(get("/api/eperson/registration/search/findByToken")
+                                .param("token", "ThisTokenDoesNotExist"))
+                   .andExpect(status().isNotFound());
+    }
+
+    private void createTokenForEmail(String email) throws Exception {
+        List<RegistrationData> registrationDatas;
+        ObjectMapper mapper = new ObjectMapper();
+        RegistrationRest registrationRest = new RegistrationRest();
+        registrationRest.setEmail(email);
+        getClient().perform(post("/api/eperson/registrations")
+                                .content(mapper.writeValueAsBytes(registrationRest))
+                                .contentType(contentType))
+                   .andExpect(status().isCreated());
```

Comment:
Needs to cleanup after POST. See https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide#CodeTestingGuide-Cleaninguptestdata

Reply:
@tdonohue There's no builder for this, and the POST doesn't return an ID, so I think we'll need to keep the current cleanup (which is happening correctly)

Reply:
yes I agree with @benbosman